ultravox/tools/wrappers.py

```diff
@@ -0,0 +1,59 @@
+import hashlib
+import json
+import os
+
+from typing import Optional
+import openai
+from ultravox.tools import tts
+
+
+class CachingChatWrapper:
+    def __init__(self, client: openai.Client, base_url: str):
+        super().__init__()
+        self._client = client
+        self._base_path = os.path.join(
+            ".cache/ds_tool/textgen", base_url.split("://", 1)[-1].replace("/", "__")
+        )
+        os.makedirs(self._base_path, exist_ok=True)
+
+    def chat_completion(self, **kwargs) -> str:
+        text_hash = hashlib.sha256(json.dumps(kwargs).encode()).hexdigest()
+
+        cache_path = os.path.join(self._base_path, f"{text_hash}.txt")
+
+        if os.path.exists(cache_path):
+            with open(cache_path, "r") as f:
+                return f.read()
+
+        response = self._client.chat.completions.create(**kwargs)
+        text = response.choices[0].message.content
+
+        with open(cache_path, "w") as f:
+            f.write(text)
+
+        return text
+
+
+class CachingTtsWrapper:
```

Comment:
I had to add a cache for text generation as well, so I ended up moving both caching wrappers into the same file and out of `tts.py`.